# Correct installed ambiguous host projections and preserve original gene evidence


66.04 second partition: withdraw only 39 verified ambiguous rhoptry protein projections, retain original gene scores and accession alternatives, and write a per-cell withdrawal ledger. Require exact original source scores and unique reviewed endpoints before changing anything. Preserve every host row and unrelated cell; retain exact TSV serialization of all unambiguous cache rows. Archive previous installed inputs externally. No inferred absence or numeric zero replaces a withheld value.

In [1]:
from scripts.correct_host_symbol_projections import migrate
summary=migrate('results/host_symbol_mapping_2026_10_08','/media/carruthers/mnt3/claude/toxoplasma_projects/datasets/installed_revision_backups/66_04_2026_10_08','results/host_symbol_mapping_2026_10_08/migration')
summary

{'withdrawn_protein_projections': 39, 'withdrawn_cells': 117, 'withdrawn_source_vs_legacy_decimal_encodings': 13, 'preserved_gene_records': 20010, 'preserved_host_rows': 20989, 'unambiguous_protein_values': 18700, 'unchanged_parasite_inference_inputs': True, 'unchanged_mouse_table': True, 'calibration_scope': 'no strategy, parasite feature/graph, track-record or claim input changed'}

All 20,010 source gene records remain available. Reduced protein coverage corrects a false entity assignment; it does not erase a gene experiment. Historical non-loss counts must reconcile through the source-verified ledger. Parasite feature/graph, track-record/claim and mouse inputs are byte-identical; no existing calibration is refitted or newly claimed. Run affected host/deposit/provenance/leakage/layout/catalogue and unchanged-inference checks before item completion.